### ISSUE TITLE
Certbot apache module - how to have different SSL vhosts file with dummy cert

### ISSUE THREAD

**Issue by jpluimers** [certbot/certbot#3771]
For some if my configurations I want the SSL/TLS Apache config files to be very different from the plain HTTP ones.

The trick at https://github.com/certbot/certbot/issues/1266#issuecomment-223744069 to specify placeholders 

```
    SSLEngine on
    SSLCertificateFile    /etc/ssl/certs/ssl-cert-snakeoil.pem
    SSLCertificateKeyFile /etc/ssl/private/ssl-cert-snakeoil.key
```

Has stopped working as you now get an error loke this:

```
# ./certbot-auto install --noninteractive --apache --domains deadlockempire.4delphi.com --cert-path /etc/letsencrypt/live/deadlockempire.4delphi.com/fullchain.pem --fullchain-path /etc/letsencrypt/live/deadlockempire.4delphi.com/fullchain.pem --key-path /etc/letsencrypt/live/deadlockempire.4delphi.com/privkey.pem
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Error while running apache2ctl configtest.

AH00526: Syntax error on line 45 of /etc/apache2/vhosts.d/deadlockempire.4delphi.com-ssl.conf:
SSLCertificateFile: file '/etc/ssl/certs/ssl-cert-snakeoil.pem' does not exist or is empty

The apache plugin is not working; there may be problems with your existing configuration.
The error was: MisconfigurationError("Error while running apache2ctl configtest.\n\nAH00526: Syntax error on line 45 of /etc/apache2/vhosts.d/deadlockempire.4delphi.com-ssl.conf:\nSSLCertificateFile: file '/etc/ssl/certs/ssl-cert-snakeoil.pem' does not exist or is empty\n",)
```

This is a bit of an issue as now there seems to be a chicken-and-egg problem:

1. at the time of generating those SSL/VHOST files you do not know the exact certificate locations
2. certbot needs to have valid certificate locations in order to insert its own locations

What would be a good way to work around this?

--jeroen

**Comment by pde** [comment]
Is `/etc/ssl/certs/ssl-cert-snakeoil.pem` actually non-existent or empty? Do you have permissions to read it?

**Comment by jpluimers** [comment]
It's a non-existent file on an opensuse based system. I'm root.

**Comment by joohoi** [comment]
Closing this as apache configuration problem. The configuration above would result in to apache restart failing even if Certbot is completely removed from the equation. The root problem this solution was trying to fix is addressed in issue #3358 so let's continue there.
